examples/medplum-fhircast-demo/src/components/PACS.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+import { BASE_URL } from '../config';
+import { usePrevious } from '../hooks';
+import { FHIRcastMessagePayload } from '../utils';
+import TopicGenerator from './TopicGenerator';
+
+function createFHIRcastMessagePayload(topic: string, patientId: string): FHIRcastMessagePayload {
+  if (!topic) {
+    throw new Error('Must provide a topic!');
+  }
+  return {
+    timestamp: new Date().toISOString(),
+    id: crypto.randomUUID(),
+    event: {
+      'hub.topic': topic,
+      'hub.event': 'patient-open', // TODO: Hardcoded for now
+      context: [
+        {
+          key: 'patient',
+          resource: {
+            resourceType: 'Patient',
+            id: patientId,
+            identifier: [
+              {
+                type: {
+                  coding: [
+                    {
+                      system: 'http://terminology.hl7.org/CodeSystem/v2-0203',
+                      value: 'MR',
+                      display: 'Medical Record Number',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        },
+      ],
+    },
+  };
+}
+
+export default function PACS(): JSX.Element {
+  const [baseUrl, setBaseUrl] = useState(BASE_URL);
+  const [baseUrlInput, setBaseUrlInput] = useState(BASE_URL);
+  const [driving, setDriving] = useState(false);
+  const [topic, setTopic] = useState<string | null>(null);
+  const [currentPatientId, setCurrentPatientId] = useState<string | null>(null);
+  const prevTopic = usePrevious(topic);
+
+  useEffect(() => {
+    // if had prev topic, and now we don't, setDriving false
+    // if didn't have and now we do, setDriving true
+    if (typeof prevTopic !== typeof topic) {
+      setDriving(!!topic);
```

Comment:
Is there a reason we do this here in another `useEffect` call rather than directly when calling `setTopic`?

Reply:
I think it's generally the React way to let state driving changes via effects when there is a condition involved. In this case we only want to update `driving` if the current `topic` type has changed from `string` to `undefined` or vice versa when compared with the `prevTopic`. 

If it seems unclear we can change it but generally the "React way" is to let the state drive the changes